lms/djangoapps/courseware/views/views.py

```diff
@@ -1724,11 +1724,13 @@ def render_xblock(request, usage_key_string, check_if_enrolled=True):
         #  masquerading as learners would often be denied access, since course
         #  staff are generally not enrolled, and viewing a course generally
         #  requires enrollment.)
-        _course_masquerade, request.user = setup_masquerade(
+        _course_masquerade, masquerade_user = setup_masquerade(
             request,
             course_key,
             staff_access,
         )
+        masquerading_as_specific_student = is_masquerading_as_specific_student(request.user, course_key)
+        request.user = masquerade_user
```

Comment:
You don't need the above dance - `is_masquerading_as_specific_student` can be called on either the original request user or the returned user from `setup_masquerade`.